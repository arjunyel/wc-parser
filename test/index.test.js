const path = require("path");
const { parse } = require("..");

describe("output", () => {
  it("html", () => {
    expect(parse(path.join(__dirname, "src/index.html"))).toMatchSnapshot();
  });

  it("complex sfc", () => {
    expect(parse(path.join(__dirname, "src/world.html"))).toMatchSnapshot();
  });

  it("simple sfc", () => {
    expect(parse(path.join(__dirname, "src/hello.html"))).toMatchSnapshot();
  });
});
