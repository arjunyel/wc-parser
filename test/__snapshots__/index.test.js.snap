// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output complex sfc 1`] = `
Object {
  "body": "import {
  props,
  withProps
} from \\"./../../node_modules/skatejs/esnext/with-props.js\\";
import { withRender } from \\"./../../node_modules/skatejs/esnext/with-render.js\\";
import { html, render } from \\"./../../node_modules/lit-html/lit-html.js\\";
function withLitHtml(Base) {
  return class extends withRender(withProps(Base || HTMLElement)) {
    rendererCallback(renderRoot, renderCallback) {
      render(renderCallback(), renderRoot);
    }
  };
}
const Base = withLitHtml();

export default class XF3327e6 extends Base {
  renderCallback({ yell }) {
    return html\`
  \${yell ? html\`<strong><slot></slot></strong>\` : html\`<em><slot></slot></em>\`}
\`;
  }
}

XF3327e6.props = { yell: props.boolean || boolean };

customElements.define(\\"x-f3327e6\\", XF3327e6);
",
  "sfc": true,
}
`;

exports[`output html 1`] = `
Object {
  "body": "<script type=\\"module\\">
import XHello from \\"./hello.js\\";
</script>
<x-8a66b12 name=\\"world\\"></x-8a66b12>",
  "sfc": false,
}
`;

exports[`output simple sfc 1`] = `
Object {
  "body": "import XWorld from \\"./world.js\\";

function ucfirst(str) {
  return str[0].toUpperCase() + str.substring(1);
}

import {
  props,
  withProps
} from \\"./../../node_modules/skatejs/esnext/with-props.js\\";
import { withRender } from \\"./../../node_modules/skatejs/esnext/with-render.js\\";
import { html, render } from \\"./../../node_modules/lit-html/lit-html.js\\";
function withLitHtml(Base) {
  return class extends withRender(withProps(Base || HTMLElement)) {
    rendererCallback(renderRoot, renderCallback) {
      render(renderCallback(), renderRoot);
    }
  };
}
const Base = withLitHtml();

export default class X8a66b12 extends Base {
  renderCallback({ name }) {
    return html\`
  Hello,
  <x-f3327e6>\${ucfirst(name)}</x-f3327e6>!
\`;
  }
}

X8a66b12.props = { name: props.string || string };

customElements.define(\\"x-8a66b12\\", X8a66b12);
",
  "sfc": true,
}
`;
